src/app/icd/server/ICDMonitoringTable.cpp

```diff
@@ -140,7 +144,20 @@ CHIP_ERROR ICDMonitoringEntry::SetKey(ByteSpan keyData)
     Crypto::Symmetric128BitsKeyByteArray keyMaterial;
     memcpy(keyMaterial, keyData.data(), sizeof(Crypto::Symmetric128BitsKeyByteArray));
 
-    // TODO - Add function to set PSA key lifetime
+#ifdef CONFIG_CHIP_CRYPTO_PSA
+    if (persistent)
+    {
+        ReturnErrorOnFailure(Crypto::FindFreeKeySlotInRange(aesKeyHandle.AsMutable<psa_key_id_t>(),
+                                                            to_underlying(Crypto::KeyIdBase::ICDAesKeyRangeStart),
+                                                            Crypto::kMaxICDClientKeys));
+        ReturnErrorOnFailure(Crypto::FindFreeKeySlotInRange(hmacKeyHandle.AsMutable<psa_key_id_t>(),
+                                                            to_underlying(Crypto::KeyIdBase::ICDHmacKeyRangeStart),
+                                                            Crypto::kMaxICDClientKeys));
+    }
```

Comment:
This seems like it shouldn't be exposed within the ICDMonitoringTable but rather in the createKey API. 
The goal of the ICDMonitoringTable was to keep as agnostic as possible of the underlying crypto implementation.
In other words, we want to avoid as much as possible `#ifdef CONFIG_CHIP_CRYPTO_PSA` which in this case can be done by improving the Keystore APIs.

Reply:
anwsering this and partially previous comment (https://github.com/project-chip/connectedhomeip/pull/34925#discussion_r1713639169):
I think we could create a separate API for setting the persistence. the limitation I had with that approach is the fact that i.e `Aes128KeyHandle` is used for other cryptographic operations as well. Because of that, we cannot associate it with key slot ranges as it won't be generic. I guess the solution for that would be creating ICD- specific key handles, and add overloads of `CreateKey` APIs for these specific handles that would handle key range and setting the persistence as well. The only thing that worries me is the fact that it would look good for two keys, but the solution is not really scalable- adding more PSA persistent keys in the future will result with lots of key-specific handles.

Reply:
Changed the approach to make ICD layer agnostic to crypto